src/FwdState.cc

```diff
@@ -503,6 +505,7 @@ FwdState::unregister(Comm::ConnectionPointer &conn)
     comm_remove_close_handler(conn->fd, closeHandler);
     closeHandler = NULL;
     serverConn = NULL;
+    destinationReceipt = nullptr;
```

Comment:
Should we do that?
For example, unregister() is called only for opened connections, e.g., from FwdState::complete(). This means that destinationReceipt may not be cleared if serverConn is closed already. So, to be consistent, we should probably either always clean destinationReceipt in such methods as FwdState::complete(), or rely on its cleaning before its usage, i.e., in FwdState::connectStart().

Reply:
I agree that there is some inconsistency in how FwdState nullifies serverConn. However, I would keep addressing that inconsistency out of scope. If our changes can work when tied to existing serverConn nullification code, I would keep them tied.